### PR TITLE
Ui finance bank details

### PR DESCRIPTION
**Pull request by MarkC33**
Fixes [AB#10612](https://einnovationteam-mlcsu.visualstudio.com/34fed099-e327-491e-a91e-4822ff25db86/_workitems/edit/10612)

Allow finance users to add a VSR number to a doctor when the claim status is 'Query'

### CODE REVIEW

File: Fmas12d.Api/RequestModels/VsrUpdate.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+using Fmas12d.Business.Models;
+
+namespace Fmas12d.Api.RequestModels
+{
+    public class VsrUpdate
```
Comment (neildavies11) [inline Fmas12d.Api/RequestModels/VsrUpdate.cs]: Change the name to VsrNumberPut so it keeps the same(ish) naming convention for the request models

File: Fmas12d.Business/Services/UserService.cs [hunk 2]
```diff
@@ -222,6 +223,37 @@ string token
...
+          VsrNumber = vsrDetails.VsrNumber,
+          ModifiedByUserId = _userClaimsService.GetUserId(),
+          ModifiedAt = DateTimeOffset.Now,
+          IsActive = true
+        };
+        _context.Add(bankDetail);
+      } else {
+        entity.VsrNumber = vsrDetails.VsrNumber;
+        entity.ModifiedByUserId = _userClaimsService.GetUserId();
+        entity.ModifiedAt = DateTimeOffset.Now;
```
Comment (neildavies11) [inline Fmas12d.Business/Services/UserService.cs]: See below (251)

File: Fmas12d.Business/Services/UserService.cs [hunk 3]
```diff
@@ -222,6 +223,37 @@ string token
...
+      }
+
```
Comment (neildavies11) [inline Fmas12d.Business/Services/UserService.cs]: Using the base class's UpdateModified method will set the modified fields for you.

File: Fmas12d.Business/Services/UserService.cs [hunk 4]
```diff
@@ -222,6 +223,37 @@ string token
       return true;
     }
 
+    public async Task<User> UpdateVsrAsync(
+      VsrUpdate vsrDetails
```
Comment (neildavies11) [inline Fmas12d.Business/Services/UserService.cs]: I'd change this to:
VsrNumberUpdate vsrNumberUpdate
Changing the local variable name so its different to the type name can be confusing

File: Fmas12d.Business/Services/UserService.cs [hunk 5]
```diff
@@ -222,6 +223,37 @@ string token
...
+      .Where(bd => bd.CcgId == vsrDetails.CcgId)
+      .SingleOrDefaultAsync();
+
+      if (entity == null) {
+        Entities.BankDetail bankDetail = new Entities.BankDetail(){
+          UserId = vsrDetails.UserId,
+          CcgId = vsrDetails.CcgId,
+          VsrNumber = vsrDetails.VsrNumber,
+          ModifiedByUserId = _userClaimsService.GetUserId(),
+          ModifiedAt = DateTimeOffset.Now,
```
Comment (neildavies11) [inline Fmas12d.Business/Services/UserService.cs]: See below (251)

File: Fmas12d.Business/Services/UserService.cs (hunk 4 above)
Comment (MarkC33) [reply Fmas12d.Business/Services/UserService.cs]: Changed to "VsrNumberUpdate model" to keep inline with other similar code

File: Fmas12d.Api/RequestModels/VsrUpdate.cs (hunk 1 above)
Comment (MarkC33) [reply Fmas12d.Api/RequestModels/VsrUpdate.cs]: Updated

File: Fmas12d.Business/Services/UserService.cs (hunk 2 above)
Comment (MarkC33) [reply Fmas12d.Business/Services/UserService.cs]: rewritten to use UpdateModified

**Pull request merged**